docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+data.
+
+All we have to do is override ``get_context_data()`` and set up the instance
+variables before making the ``super()`` call. This may seem a bit unusual
+because these variables are normally set up in ``get()`` or ``post()``, but in
+this case they have to be set up in both, and ``get_context_data()`` is the
+common execution path we can use to avoid duplicate logic.
+
+You may wish to access the object in ``form_valid()`` (e.g. for redirection);
+if you do, you have to use the ``get_object()`` method because
+``get_context_data()`` is not used in this execution path.
+
+(As a general note, it's good to remember the difference between
+``form_valid()`` and ``form_invalid()`` whenever you're working with FormView:
+though similarly named, their logic is completely different and it's useful to
+think of ``form_invalid()`` as 'retry get()' method).
+
+.. snippet::
+    :filename: views.py
+
+    class AuthorBooksView(SingleObjectMixin, MultipleObjectMixin, FormView):
+        model = Author
+        form_class = AuthorInterestForm
+        paginate_by = 2
+        template_name = "author-books.html"
+
+        def get_context_data(self, **kwargs):
+            self.object = self.get_object()
+            self.object_list = self.object.books.all()
+            return super(AuthorBooksView, self).get_context_data(**kwargs)
+
+        def form_valid(self, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect("author-books", pk=self.get_object().pk)
+
+
+.. code-block:: html+django
+
+    <h2>Author: {{ author.name }}</h2>
+
+    <ol>
+      {% for book in page_obj %}
+        <li>{{ book.name }}</li>
+      {% endfor %}
+    </ol>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ form.as_table }}
+        </table>
+    <p><input type="submit" name="author_form" value="Submit" /></p>
+    </form>
+
+Combined Create and List Views
+==============================
+
+This view is quite simple: we're just pulling in the listing from
+``MultipleObjectMixin.get_context_data()``:
+
+.. snippet::
+    :filename: views.py
+
+    from django.views.generic import CreateView
+
+    class BooksCreateView(MultipleObjectMixin, CreateView):
+        """
+        We're overriding get_context_data() to set up `self.object` and `self.object_list` to
+        be used in respective parent get_context_data() methods.
+
+        Note: we cannot rely on ListView.get() to create `self.object_list` because we also need it on POST,
+        when form_invalid() calls get_context_data().
+        """
+        model         = Book
+        paginate_by   = 10
+        fields        = ["name", "author"]
+        success_url   = reverse_lazy("list-books-create")
+        template_name = "books.html"
+
+The template is the same ``books.html`` we've used before.
+
+Formset View
+============
```

Comment:
For a simple example like this it doesn't even need documenting separately (for a model formset for example a formview will work!) In reality, this creates a lot of inflexibility in the creation of the formset. Also it doesn't use `get_form_class` sensibly.